docs/design/2026-04-25-single-flow-owner-design.md

```
@@ -0,0 +1,120 @@
+### Saving
+
+Track modified files by maintaining dirty flags per `Route`. When saving,
+iterate all dirty routes and serialize each `FlowDefinition` or `FunctionDefinition`
+to its file (determined by `source_url`).
+
+Serialization methods for `FlowDefinition` and `FunctionDefinition` need to be
+added to flowcore, moving logic currently in flowedit's `file_ops.rs`.
+
+### Data Flow Diagram
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)